src/core/unit-printf.h

```diff
@@ -1,7 +1,22 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+#include "env-util.h"
+#include "fileio.h"
 #include "unit.h"
 
 int unit_name_printf(const Unit *u, const char* text, char **ret);
-int unit_full_printf(const Unit *u, const char *text, char **ret);
+int unit_full_printf_full(const Unit *u, const char *text, size_t max_length, char **ret);
+static inline int unit_full_printf(const Unit *u, const char *text, char **ret) {
+        return unit_full_printf_full(u, text, LONG_LINE_MAX, ret);
+}
+static inline int unit_path_printf(const Unit *u, const char *text, char **ret) {
+        return unit_full_printf_full(u, text, PATH_MAX-1, ret);
+}
+static inline int unit_fd_printf(const Unit *u, const char *text, char **ret) {
+        return unit_full_printf_full(u, text, 255, ret);
```

Comment:
maybe introduce a new macro for the length, and use this here and in fdname_is_valid(). Maybe called FDNAME_MAX?